### PR TITLE
Implement  ovh_iploadbalancing_failover_ip_attach resource/data

### PR DESCRIPTION
**Pull request by bchaillou-ovh**
# Description

This PR is a request from Webhosting Infra team:
- Implement ovh_iploadbalancing_failover_ip_attach resource/data

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [X] Documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [X] Test A: `make testacc TESTARGS="-run TestAccDataSourceIpLoadbalancingFailoverIpAttach"`
- [X] Test B: `make testacc TESTARGS="-run TestAccResourceIpLoadbalancingFailoverIpAttach"`

**Test Configuration**:
- Terraform version: `terraform version`: Terraform v1.2.6
- Use credentials (`OVH_APPLICATION_KEY`, `OVH_APPLICATION_SECRET`, `OVH_CONSUMER_KEY`) containing as resources:
   - an iplb with ip failover attached
   - an ip failover not attached
- For Test A:
   - set `OVH_IPLB_SERVICE_TEST` with iplb name
   - set `OVH_IP_BLOCK_TEST` with ip failover attached
- For Test B:
   - set `OVH_IPLB_SERVICE_TEST` with iplb name
   - set `OVH_IP_BLOCK_TEST` with ip failover not attached
- Existing HCL configuration you used:

```hcl
data "ovh_iploadbalancing_failover_ip_attach" "failoverip" {
   service_name = "loadbalancer-xxxxxxxxxxxxxxxxxx"
   ip = "x.x.x.x/y"
}

resource "ovh_iploadbalancing_failover_ip_attach" "move_failoverip" {
   service_name = "loadbalancer-xxxxxxxxxxxxxxxxxx"
   ip = "x.x.x.x/y"
   to = "loadbalancer-xxxxxxxxxxxxxxxxxx"
}
```

# Checklist:
- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings or issues
- [X] I have added acceptance tests that prove my fix is effective or that my feature works
- [X] New and existing acceptance tests pass locally with my changes
- [X] I ran succesfully go mod vendor if I added or modify go.mod file

### CODE REVIEW

**Review by scraly — CHANGES_REQUESTED** [review]
When you create resources and/or datasources, please create associated documentation pages.

File: website/docs/d/iploadbalancing_failover_ip_attach_html.markdown [hunk 1]
```diff
@@ -0,0 +1,27 @@
+---
+subcategory : "Load Balancer (IPLB)"
+---
+
+# ovh_iploadbalancing_vrack_network (Data Source)
```
Comment (scraly) [inline website/docs/d/iploadbalancing_failover_ip_attach_html.markdown:5]: What is the correct name of the documetation page? :-) 

"ovh_iploadbalancing_failover_ip_attach" ? ;-)

File: website/docs/d/iploadbalancing_failover_ip_attach_html.markdown [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+Use this data source to get list failover ip routed for your IPLoadbalancer associated with your OVHcloud account.
+
+## Example Usage
+
+```hcl
+data ovh_iploadbalancing_failover_ip_attach failoverip {
```
Comment (scraly) [inline website/docs/d/iploadbalancing_failover_ip_attach_html.markdown:12]:
```suggestion
data "ovh_iploadbalancing_failover_ip_attach" "failoverip" {
```

File: website/docs/r/iploadbalancing_failover_ip_attach.html.markdown [hunk 3]
```diff
@@ -0,0 +1,32 @@
+---
+subcategory : "Load Balancer (IPLB)"
+---
+
+# ovh\_iploadbalancing\_failover\_ip\_attach
```
Comment (scraly) [inline website/docs/r/iploadbalancing_failover_ip_attach.html.markdown:5]:
```suggestion
# ovh_iploadbalancing_failover_ip_attach
```

File: website/docs/r/iploadbalancing_failover_ip_attach.html.markdown [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+Attach an IP for a loadbalancer service
+
+## Example Usage
+
+```hcl
+resource ovh_iploadbalancing_failover_ip_attach failoverip {
```
Comment (scraly) [inline website/docs/r/iploadbalancing_failover_ip_attach.html.markdown:12]:
```suggestion
resource "ovh_iploadbalancing_failover_ip_attach" "failoverip" {
```

File: website/docs/r/iploadbalancing_failover_ip_attach.html.markdown [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `service_name` - (Required) The internal name of your IP load balancing
+* `ip` - (Required) IP to move
+* `to` - (Required) Service distination
```
Comment (scraly) [inline website/docs/r/iploadbalancing_failover_ip_attach.html.markdown:25]:
```suggestion
* `to` - (Required) Service destination
```

File: website/docs/d/iploadbalancing_failover_ip_attach_html.markdown [hunk 6]
```diff
@@ -0,0 +1,27 @@
+---
+subcategory : "Load Balancer (IPLB)"
+---
+
+# ovh_iploadbalancing_vrack_network (Data Source)
+
+Use this data source to get list failover ip routed for your IPLoadbalancer associated with your OVHcloud account.
```
Comment (scraly) [inline website/docs/d/iploadbalancing_failover_ip_attach_html.markdown:7]:
```suggestion
Use this data source to list all failover IP routed to this IPLB.
```

**Review by scraly — CHANGES_REQUESTED** [review]
Several typos found + please list acceptance tests to run and test them please :)

File: ovh/types_iploadbalancing.go [hunk 7]
```diff
@@ -669,3 +674,72 @@ func (opts *IpLoadbalancingVrackNetworkUpdateOpts) FromResource(d *schema.Resour
...
+	IpTaskStatusDoing         IpTaskStatus = "doing"
+	IpTaskStatusDone          IpTaskStatus = "done"
+	IpTaskStatusInit          IpTaskStatus = "init"
+	IpTaskStatusOvhError      IpTaskStatus = "ovhError"
+	IpTaskStatusTodo          IpTaskStatus = "todo"
+)
+
+type IpTask struct {
+	Comment     string       `json:"comment"`
+	Destination string       `json:"destination"`
```
Comment (amstuta) [inline ovh/types_iploadbalancing.go:692]: from what I see in the API schemas here: https://api.ovh.com/console-preview/?section=%2Fip&branch=v1#post-/ip/-ip-/move
the field `destination` is an object containing a `serviceName` field

File: ovh/types_iploadbalancing.go [hunk 8]
```diff
@@ -669,3 +674,72 @@ func (opts *IpLoadbalancingVrackNetworkUpdateOpts) FromResource(d *schema.Resour
...
+	DoneDate    time.Time    `json:"doneDate"`
+	Function    string       `json:"function"`
```
Comment (amstuta) [inline ovh/types_iploadbalancing.go:694]: you should declare a const for this ike you did with `IpTaskStatus`

File: ovh/types_iploadbalancing.go [hunk 9]
```diff
@@ -669,3 +674,72 @@ func (opts *IpLoadbalancingVrackNetworkUpdateOpts) FromResource(d *schema.Resour
...
+	LastUpdate  time.Time    `json:"lastUpdate"`
+	StartDate   time.Time    `json:"startDate"`
+	Status      IpTaskStatus `json:"status"`
+	TaskID      int64        `json:"taskId"`
+}
+
+type IpMoveCreateOpts struct {
+	Nexthop string `json:"nexthop"`
```
Comment (amstuta) [inline ovh/types_iploadbalancing.go:702]:
```suggestion
	Nexthop string `json:"nexthop,omitempty"`
```

**Review by amstuta — CHANGES_REQUESTED** [review]

**Comment by amstuta** [comment]
Closing this PR for inactivity, don't hesitate to re-open it if still needed.

**Pull request closed without merging**